### ISSUE TITLE
Clarify MSRV policy

### ISSUE THREAD

**Issue by svartalf** [KokaKiwi/rust-hex#39]
Latest `0.4.1` release [broke](https://github.com/heim-rs/heim/runs/444776330) support for older Rust versions compared to `0.4.0`, for example, accessing enum variants via `Self` were not available for Rust 1.36.0:

https://github.com/KokaKiwi/rust-hex/blob/78359a859c356a6c756cccad16977fdedc4824c5/src/error.rs#L26-L30

In my case this version of `hex` were used as a dependency in the `Cargo.toml`,
```toml
[dependencies]
hex = "0.4.0"
```

and after the new release, cargo automatically started using `0.4.1` as they are [assumed to be compatible](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio).

Generally it seems unexpected to update MSRV (Minimal Supported Rust Version) in the [patch](https://semver.org/) versions, so it would be nice to clarify your policy on the compiler versions support.
Usual options are either to test specific Rust version [in the CI workflow](https://github.com/actions-rs/meta/blob/master/recipes/msrv.md) or to put a note in the README file that this project is compatible only with the latest stable Rust version.

**Comment by KokaKiwi** [comment]
Oh shoot indeed, i should've been more careful releasing the 0.4.1 as i didn't think about the used Rust's version (only thought about the API stability as per semver requirements, which is okay but not always enough indeed) :x

As the breaking stuff seems to be easily fixable, i think i'll release a non-breaking corrective version (0.4.2) and yank the 0.4.1 version (to avoid other broken stuff), and stating in the README.md a MSRV for at least an old enough Rust version (maybe 1.36.0 actually) as this lib should be mostly simple and thus wouldn't require an "edging" Rust version
Along this statement i'll add the MSRV to the Rust's version matrix in the CI so it'd be checked from now on to avoid any future breaking version like this...

PS: also if it's okay for you, i'll gladly do with some feedback about this plan of doing as i never dealt with yank[ed/ing] stuff before, so any tip or comment on this would be greatly valuable for me ^-^

**Comment by svartalf** [comment]
Yeah, unfortunately there is no builtin way to force MSRV right now ([yet](https://github.com/rust-lang/rust/issues/65262)), so at least you can utilize [matrix builds](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy) to check both for `stable` and some MSRV version.

Also I'm in no position to ask about some specific Rust version, but considering the amount of the [dependant crates and repositories](https://libraries.io/cargo/hex), there is be a chance that sticking with Rust 1.36 will break compatibility to someone.
Naive and dirty way to determine the minimal version would be a bunch of `rustup target install {version} && cargo +{version} build` commands issued on `0.4.0` tag and going back to some extremely old Rust version, and I fail to find better option for now :(

Crates yanking is pretty easy, the following command should do the thing:
```
cargo yank --vers 0.4.1 hex
```
Note that `yank` subcommand also has the `--undo` flag, so in case of any typos it will be easy to rollback it back to the "un-yanked" state.

**Comment by msdrigg** [comment]
I just wanted to follow up on this issue. I checked your package using this library https://github.com/foresterre/cargo-msrv

Here are the results for your package. To get each result, I ran `git checkout <tag>`, and then `cargo msrv --bisect`

* For Hex v0.4.3, the msrv is 1.46.0
* For Hex v0.4.2, the msrv is 1.31.1
* For Hex v0.4.1, the msrv is 1.37.0
* For Hex v0.4.0, the msrv is 1.31.1
* For Hex v0.3.2 and all lower, the msrv is 1.17.0
